### PR TITLE
Chromium: Implement basic navigation using URL bar

### PR DESCRIPTION
**Pull request by zakharvoit**
Implement basic navigation methods using WebContents' navigation controller. This allows the browser's URL bar (including back, forward and reload buttons) to work with the Chromium backend. Method getUriFromString() is reused from the weblayer based implementation.

### CODE REVIEW

File: app/src/common/chromium/com/igalia/wolvic/browser/api/impl/SessionImpl.java [hunk 1]
```diff
@@ -52,7 +60,47 @@ private void init() {
...
+        if (!mIsDisplayAcquired) {
+            mInitialUri = uri;
+            return;
+        }
+
+        LoadUrlParams params = new LoadUrlParams(getUriFromString(uri).toString());
+        mNavigationController.loadUrl(params);
+    }
+
+    private Uri getUriFromString(@NonNull String str) {
```
Comment (jdapena) [inline app/src/common/chromium/com/igalia/wolvic/browser/api/impl/SessionImpl.java:72]: Do you really need this? LoadUrlParams accepts a string, and will internally process it following Chromium heuristics (with exception of the case covered by sanitizeUrl that we cover below with www. prefix).

What is interesting is adding a way to prepopulate to search engine when the url does not match a valid pattern. To do that, we can use same logic used in weblayer TopBarImpl.java:
* It parses with Uri.parse.
* If it is absolute, then it passes the string.
* Then it checks if domain is valid and adds https://
* Only then, if it is not a valid domain nor absolute URI, then it creates the google search query.

See here: https://github.com/chromium/chromium/blob/4bb364edffee3e741028942127a27a2afcc18ce4/weblayer/shell/android/webengine_shell_apk/src/org/chromium/webengine/shell/TopBarImpl.java#L66

Comment (zakharvoit) [reply app/src/common/chromium/com/igalia/wolvic/browser/api/impl/SessionImpl.java:72]: It seems like URL validation is not performed by NavigationController. So we need this, see for example https://source.chromium.org/chromium/chromium/src/+/main:android_webview/java/src/org/chromium/android_webview/AwContents.java;l=2250;drc=bec0ef46974cd9a6bc0f1f1ae34baa11913f77af

I can suggest reusing the Chromium's logic by using https://source.chromium.org/chromium/chromium/src/+/main:components/url_formatter/android/java/src/org/chromium/components/url_formatter/UrlFormatter.java
what do you think?

For that I will need to add another 1-line CL to the wolvic-chromium, but it seems like a more reliable approach.

Comment (jdapena) [reply app/src/common/chromium/com/igalia/wolvic/browser/api/impl/SessionImpl.java:72]: Yes, it looks good to me.

**Review by svillar — CHANGES_REQUESTED** [review]
Mind rebasing on top of main?

**Comment by zakharvoit** [comment]
depends on https://github.com/Igalia/wolvic-chromium/pull/8

**Review by jdapena — APPROVED** [review]

**Review by duswnchl — APPROVED** [review]

**Review by mshin-wolvic — APPROVED** [review]

**Pull request merged**